Inc/ST-LIB_HIGH/Protections/Boundary.hpp

```diff
@@ -16,246 +22,224 @@ enum ProtectionType : uint64_t {
-			external_pointer(&external_pointer){
-		external_pointer = this;
-	};
-	Boundary(double* src, Boundary<double, Protector> boundary): src(src),bound(boundary.bound),time_limit(boundary.time_limit),frequency(boundary.frequency),moving_order(frequency*time_limit/100), external_pointer(boundary.external_pointer){
-		*external_pointer = this;
-	}
-	Boundary(double* src, double bound ,float time_limit, float frequency): src(src),bound(bound) ,time_limit(time_limit), frequency(frequency),moving_order(frequency*time_limit/100), external_pointer(nullptr){}
-
-	Boundary<double,Protector>** external_pointer;
-
-private:
-	MeanCalculator<100> mean_calculator;
-	vector<double> mean_moving_average;
-	uint16_t moving_order = 0;
-	uint16_t moving_last = -1;
-	uint16_t moving_first = 0;
-	uint16_t moving_counter = 0;
-	double accumulator = 0;
-	bool still_good = true;
-public:
-	bool check_accumulation(double value){
-		if(!still_good) return false;
-		mean_calculator.input(abs(value));
-		if(mean_calculator.output_value == 0){
-			return true;
-		}
-		mean_calculator.reset();
-		if(moving_counter < moving_order) {
-			moving_last++;
-			mean_moving_average[moving_last] = mean_calculator.output_value;
-			accumulator += mean_calculator.output_value;
-			moving_counter++;
-		}
-		accumulator -= mean_moving_average[moving_first] / moving_order;
-		moving_first = (moving_first + 1) % moving_order;
-		moving_last = (moving_last + 1) % moving_counter;
-		mean_moving_average[moving_last] = mean_calculator.output_value;
-		accumulator += mean_moving_average[moving_last] / moving_order;
-		if(accumulator > bound){
-			still_good = false;
-			return false;
-		} 
-		return true;
-	}
 
-	bool check_bounds() override{
-		return still_good;
-	}
 
-	int get_string_size()override{
-		return snprintf(nullptr,0,format,to_string(*src).c_str(), to_string(time_limit).c_str());
-	}
-	char* serialize(char* dst)override{
-		sprintf(dst, format, to_string(*src).c_str(), to_string(time_limit).c_str());
-		return dst;
-	}
-};
-
-template<>
-struct Boundary<float, TIME_ACCUMULATION> : public BoundaryInterface {
-	static constexpr ProtectionType Protector = TIME_ACCUMULATION;
-	static constexpr const char* format = "\"type\": \"TIME_ACCUMULATION\", \"data\": { \"value\": %s, \"bound\": %s,\"timelimit\": %s }";
-	
-	Boundary(float bound, float time_limit, float frequency, Boundary<float, Protector>*& external_pointer): bound(bound),time_limit(time_limit) ,frequency(frequency), moving_order(frequency*time_limit/100),
+template<typename Type>
+requires(std::is_floating_point_v<Type>)
+struct Boundary<Type, TIME_ACCUMULATION> : public BoundaryInterface {
+	static constexpr ProtectionType Protector = TIME_ACCUMULATION;	
+	Boundary(Type bound, float time_limit, float frequency, Boundary<Type, Protector>*& external_pointer): bound(bound),time_limit(time_limit) ,frequency(frequency), moving_order(frequency*time_limit/100),
```

Comment:
This class is critical, test if it still works after removing specialization

Reply:
will do, but note that the previous implementations were different, one didn't call execute().